selftests/unit/utils/test_asset.py

```diff
@@ -0,0 +1,46 @@
+import os
+import tempfile
+import unittest
+
+from avocado.utils.asset import Asset
+from selftests.utils import TestCaseTmpDir, setup_avocado_loggers
+
+setup_avocado_loggers()
+
+
+class TestAsset(TestCaseTmpDir):
+    def test_check_hash_by_diferent_algorithm(self):
```

Comment:
It seems to me that this could be broken down into three tests:

* `test_has_valid_hash`
* `test_refuses_invalid_hash`
* `test_adds_valid_hash`

For the last one, AFAICT we're not protecting against adding the same hash/algorithm combination multiple times.  This is something to consider IMO.

Reply:
sure, I will split it and add the new test.